plugin/trino-sqlserver/src/main/java/io/trino/plugin/sqlserver/SqlServerClient.java

```diff
@@ -239,12 +252,39 @@ public class SqlServerClient
             }
             // Domain#simplify can turn a discrete set into a range predicate
             // Push down of range predicate for varchar/char types could lead to incorrect results
-            // when the remote database is case insensitive
+            // when the remote database is case-insensitive
             return DISABLE_PUSHDOWN.apply(session, domain);
         }
         return FULL_PUSHDOWN.apply(session, simplifiedDomain);
     };
 
+    // Dates prior to the Gregorian calendar switch in 1582 can cause incorrect results when pushed down,
+    // so we disable predicate push down when the domain contains values prior to 1583
+    private static final Instant DISABLE_DATETIMEOFFSET_PUSHDOWN_IF_BEFORE = Instant.parse("1583-01-01T00:00:00Z");
```

Comment:
The variable name gives an assumption to be like a `boolean` field or some function which returns boolean

Reply:
Changed them to `GREGORIAN_SWITCH_INSTANT` and `GREGORIAN_SWITCH_DATETIMEOFFSET`, open to other suggestions.